datalad/distribution/install.py

```diff
@@ -43,6 +42,39 @@ def _with_sep(path):
     return path + sep if not path.endswith(sep) else path
 
 
+def get_git_dir(path):
+    """figure out a repo's gitdir
+
+    '.git' might be a  directory, a symlink or a file
+
+    Parameter
+    ---------
+    path: str
+      currently expected to be the repos base dir
+
+    Returns
+    -------
+    str
+      relative path to the repo's git dir; So, default would be ".git"
+    """
+
+    from os.path import isfile
+    from os import readlink
+
+    dot_git = opj(path, ".git")
+    if not exists(dot_git):
+        raise RuntimeError("Missing .git in %s." % path)
+    elif islink(dot_git):
+        git_dir = readlink(dot_git)
+    elif isdir(dot_git):
+        git_dir = ".git"
+    elif isfile(dot_git):
+        with open(dot_git) as f:
+            git_dir = f.readline().lstrip("gitdir:").strip()
```

Comment:
incorrect lstrip -- you specified a set of characters to strip, so it would remove any of those from the left of the path.  how likely it to start with any of those -- probably impossible since would probably start with '.', but still -- somewhat "incorrect" and imho better to validate that it starts with gitdir: and strip it by `[7:]`